packages/backend/src/auth/middleware.ts

```diff
@@ -14,14 +18,28 @@ export function authenticate(
   const { token } = req.cookies
   if (token) {
     const secretKey = config.auth.jwtSecretKey
-    const verification = jwt.verify(token, secretKey)
-    if (typeof verification === 'string') {
-      next(
-        new Error(`Malformed JWT token: received 'string', expected 'object'`),
-      )
-    } else {
-      ;(req as RequestWithToken).jwtPayload = verification
-      next()
+    try {
+      const verification = jwt.verify(token, secretKey)
+      if (typeof verification === 'string') {
+        next(
+          new Error(
+            `Malformed JWT token: received 'string', expected 'object'`,
+          ),
+        )
+      } else {
+        ;(req as RequestWithToken).jwtPayload = verification
+        next()
+      }
+    } catch (error) {
+      if (error instanceof TokenExpiredError) {
+        res.status(403).json({ error: 'Your token has expired.' })
```

Comment:
Is there perhaps a way to force the cookie to be removed if the token is expired?

Reply:
Hmm, not sure, but you can look at logOut function